### PR TITLE
NJ Blind or Disabled Exemption 

### LINKED ISSUE #1923: NJ Blind or Disabled Exemption
Line 8 – Blind or Disabled
"You can claim a $1,000 exemption if you were blind or disabled on the last day of the tax year. If you are filing jointly,
your spouse can take a $1,000 exemption if they were blind or
disabled on the last day of the tax year. You cannot claim this
exemption for your domestic partner or dependents."

https://www.state.nj.us/treasury/taxation/pdf/current/1040i.pdf#page=8
https://casetext.com/statute/new-jersey-statutes/title-54a-new-jersey-gross-income-tax-act/chapter-54a3-personal-exemptions-and-deductions/section-54a3-1-personal-exemptions-and-deductions

Good place to start: https://github.com/PolicyEngine/policyengine-us/blob/e234e6e2f8a72b80566d3ef0e62719c6472f7285/policyengine_us/variables/gov/states/md/tax/income/exemptions/md_blind_exemption.py

### PR DESCRIPTION
**Pull request by ZiqianCui**
Fixes #1923

### CODE REVIEW

File: policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    def formula(tax_unit, period, parameters):
+        # First get their filing status.
+        filing_status = tax_unit("filing_status", period)
+
+        # Then get the NJ blind ir disabled exemptions part of the parameter tree.
+        p = parameters(period).gov.states.nj.tax.income.exemptions.blind
+
+        # Get the individual blind status.
+        blind_head = tax_unit("blind_head", period).astype(int)
+
```
Comment (leogoldman) [inline policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py:21]: Check for blind or disabled, so adding `disabled_head = tax_unit("disabled_head", period)` and then something like `head_eligible = blind_head | disabled_head`.

File: policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        # Determine whether spouse is eligible.
+        joint = filing_status == filing_status.possible_values.JOINT
+        blind_spouse = (tax_unit("blind_spouse", period) * joint).astype(int)
```
Comment (leogoldman) [inline policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py:24]: Will be similar for spouse: `disabled_spouse = tax_unit("disabled_spouse", period)` and then something like `spouse_eligible = blind_spouse | disabled_spouse`.

File: policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        # Calculate total blind exemption.
+        return (blind_head + blind_spouse) * p.amount
```
Comment (leogoldman) [inline policyengine_us/variables/gov/states/nj/tax/income/exemptions/nj_blind_exemption.py:27]: `return (head_eligible + spouse_eligible) * p.amount`

**Review by leogoldman — COMMENTED** [review]
This is great! Glad to see it already up and running and passing the tests. Right now, I think we're just capturing whether or not the household head (and possibly spouse) are blind, but since blind or disabled is one line in the NJ tax code, I think it makes sense that this exemption should capture both cases. I left some suggested changes to implement that based on logic in the link below.
Once it does both blind and disabled I think we can also rename to nj_blind_or_disabled_exemption.py (and then rename all the relevant folders/tests/etc.).

https://github.com/PolicyEngine/policyengine-us/blob/35da26653e0be68ce3c4834829a3da816d83c8e6/policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_taxunit_eligible.py

File: policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+description: NJ provides a blind or disabled exemption of $1k for individuals and $2k for couples.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml:1]: Descriptions omit values
```suggestion
description: New Jersey provides a blind or disabled exemption of this amount per filer or spouse.
```

File: policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: currency-USD
+  label: NJ Blind Or Disabled Exemption
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml:4]:
```suggestion
  label: New Jersey blind or disabled exemption
```

File: policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  reference: 
+    - title: New Jersey State 54A 3-1
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/exemptions/blind_or_disabled/amount.yaml:6]:
```suggestion
    - title: New Jersey State 54A 3-1(b)(5-6)
```

**Review by MaxGhenis — APPROVED** [review]
Looks great, just a couple nits I'll commit and merge. Thanks!

**Pull request merged**